Flutter/cartesian-charts/export-cartesian-chart.md

```diff
@@ -17,190 +17,189 @@ To export the Cartesian chart as a PNG image, we can get the image by calling [`
-      }
-       
-    Future<void> _renderImage() async {
-      final List<int> bytes = await _readImageData();
-      if (bytes != null) {
-        final Directory documentDirectory =
-          await getApplicationDocumentsDirectory();
-        final String path = documentDirectory.path;
-        const String imageName = 'cartesianchart.png';
-        imageCache!.clear();
-        final File file = File('$path/$imageName');
-        file.writeAsBytesSync(bytes);
-        await Navigator.of(context).push<dynamic>(
-          MaterialPageRoute<dynamic>(
-            builder: (BuildContext context) {
-              return Scaffold(
-                appBar: AppBar(),
-                body: Center(
-                  child: Container(
-                    color: Colors.white,
-                    child: Image.file(file),
-                  ),
-                ),
-              );
-            },
-          ),
-        );
-      }
+              ]
+            ),
+            TextButton(
+              child: const Text('Export as image'),
+              onPressed: () {
+                _renderChartAsImage();
+              },
+            )
+          ]
+        ),
+      );
     }
 
-      Future<List<int>> _readImageData() async {
-        final dart_ui.Image data =
-        await _chartKey.currentState!.toImage(pixelRatio: 3.0);
-        final ByteData? bytes =
-        await data.toByteData(format: dart_ui.ImageByteFormat.png);
-        return bytes!.buffer.asUint8List(bytes.offsetInBytes, bytes.lengthInBytes);
-      }
+    Future<void> _renderChartAsImage() async {
+      final ui.Image data =
+        await _cartesianChartKey.currentState!.toImage(pixelRatio: 3.0);
+      final ByteData? bytes =
+        await data.toByteData(format: ui.ImageByteFormat.png);
+      final Uint8List imageBytes =
+        bytes!.buffer.asUint8List(bytes.offsetInBytes, bytes.lengthInBytes);
+      await Navigator.of(context).push<dynamic>(
+        MaterialPageRoute<dynamic>(
+          builder: (BuildContext context) {
+            return Scaffold(
+              body: Image.memory(imageBytes)
+            );
+          },
+        ),
+      );
     }
+  }
 
 {% endhighlight %}
 
 ## Export PDF
 
 Similar to the above way, we can also export the rendered chart as a PDF document. We create the pdf document using pdf component. This can be done in the application level itself and please find the code snippet below.
+And it is necessary to include the platform-specific file generating codes to save the file as a PDF document. So, we have created two dart files ([`save_file_mobile`](https://github.com/syncfusion/flutter-examples/blob/master/lib/samples/pdf/helper/save_file_mobile.dart) and [`save_file_web`](https://github.com/syncfusion/flutter-examples/blob/master/lib/samples/pdf/helper/save_file_web.dart)) to save and launch the file in different platforms. On that two dart files, we have used a common method to save the exported content as a file.
```

Comment:
It is necessary to include the platform-specific file generating codes to save the file as a PDF document. So, create two dart files ([`save_file_mobile`](https://github.com/syncfusion/flutter-examples/blob/master/lib/samples/pdf/helper/save_file_mobile.dart) and [`save_file_web`](https://github.com/syncfusion/flutter-examples/blob/master/lib/samples/pdf/helper/save_file_web.dart)) to save and launch the file in different platforms.